tooling/nargo_fmt/src/formatter/chunks.rs

```diff
@@ -0,0 +1,776 @@
+    /// Note that if this group is a MethodCall, this is checked for the ExpressionList group
+    /// inside it.
+    pub(crate) fn has_lambda_as_last_expression_in_list(&self) -> bool {
+        self.chunks.iter().any(|chunk| {
+            if let Chunk::Group(group) = chunk {
+                if self.kind.is_method_call() && group.kind.is_expression_list() {
+                    group.has_lambda_as_last_expression_in_list()
+                } else {
+                    matches!(group.kind, GroupKind::LambdaAsLastExpressionInList { .. })
+                }
+            } else {
+                false
+            }
+        })
+    }
+
+    /// Finds the `LambdaAsLastExpressionInList` associated to this group and sets its indentation
+    /// to the given value.
+    pub(crate) fn set_lambda_as_last_expression_in_list_indentation(
+        &mut self,
+        indentation_to_set: i32,
+    ) {
+        for chunk in self.chunks.iter_mut() {
+            if let Chunk::Group(group) = chunk {
+                if self.kind.is_method_call() && group.kind.is_expression_list() {
+                    group.set_lambda_as_last_expression_in_list_indentation(indentation_to_set);
+                } else if let GroupKind::LambdaAsLastExpressionInList { indentation, .. } =
+                    &mut group.kind
+                {
+                    if indentation.is_none() {
+                        *indentation = Some(indentation_to_set);
+                    }
+                }
+            }
+        }
+    }
+
+    /// Before writing a Chunks object in multiple lines, create a new one where `TrailingComma`
+    /// is turned into `Text`. Because Chunks will glue two consecutive `Text`s together, if we
+    /// have two chunks `Text("123"), TrailingComma`, we'll consider the entire string "123,"
+    /// when deciding whether we can still write in the current line or not.
+    pub(crate) fn prepare_for_multiple_lines(self) -> ChunkGroup {
+        let mut group = ChunkGroup {
+            chunks: Vec::new(),
+            one_chunk_per_line: self.one_chunk_per_line,
+            force_multiple_lines: self.force_multiple_lines,
+            tag: self.tag,
+            kind: self.kind,
+            force_multiline_on_children_with_same_tag_if_multiline: self
+                .force_multiline_on_children_with_same_tag_if_multiline,
+        };
+
+        for chunk in self.chunks {
+            match chunk {
+                Chunk::Text(chunk) => group.text(chunk),
+                Chunk::TrailingComma => {
+                    // If there's a trailing comma after a group, append the text to that group
+                    // so that it glues with the last text present there (if any)
+                    if let Some(Chunk::Group(group)) = group.chunks.last_mut() {
+                        group.add_trailing_comma_to_last_text();
+                    } else {
+                        group.text(TextChunk::new(",".to_string()));
+                    }
```

Comment:
```suggestion
                    group.add_trailing_comma_to_last_text();
```
At least I think it's the same thing 👀

Reply:
Good point! I think I wrote this before eventually writing the other method.